app/domain/authentication/authn_gcp/validate_resource_restrictions_match_jwt.rb

```diff
@@ -0,0 +1,57 @@
+module Authentication
+  module AuthnGcp
+
+    # This class is responsible of validating resource restrictions values against JWT token.
+    # the assumption is that resource restrictions list contains only permitted types which already validated
+    # in previous steps
+    ValidateResourceRestrictionsMatchJWT = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger
+      },
+      inputs:       %i(resource_restrictions decoded_token restriction_prefix)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnGcp::ValidatingResourceRestrictionsValues.new)
+        validate
+        @logger.debug(LogMessages::Authentication::AuthnGcp::ValidatedResourceRestrictionsValues.new)
+      end
+
+      private
+
+      def validate
+        @resource_restrictions.each do |r|
+          restriction_type = r.type
+          restriction_value = r.value
+          restriction_value_from_token = restriction_value_from_token(restriction_type_without_prefix(restriction_type))
+          unless restriction_value == restriction_value_from_token
+            raise Errors::Authentication::Jwt::InvalidResourceRestrictions.new(restriction_type)
```

Comment:
do we want to print also the value in the host? do we want to print also the value in the token? 

are these considered sensitive information?

Reply:
IMO both values (in the host and in the token) are sensitive, WDYT?

Reply:
first, let's add @sashaCher to the mix

Reply:
the thing is that we assume that whoever reads the logs has access to the Conjur container. So they already have some privileges.

Now, the annotations are part of the configuration and thus are not sensitive (or else they would have been set as secrets).

Reply:
As i see at:
1. The token is equivalent to password, so in my opinion we should not print its values to the logs
2. Regarding the annotations you are right they are exists in the policy so they already exposed, but i do not like it because it is not a really annotations and more like a secrets, but we choose to implement it as annotations, so the decision here should take into consideration that those restrictions values may will be implemented in a different way in the future.

Reply:
when in doubt, do not print. let's continue without printing.

if we'll see a case where it would have helped us then we can revisit this decision.

Reply:
Annotations are not secret in general but in any case they can be accused only by authenticated Conjur user with relevant permissions only. The person that reads logs can have significantly lower permissions means may be it has only monitoring access on relevant environment.
So I think that we should no print values in logs.

@orenbm 
> when in doubt, do not print. let's continue without printing.

Good approach!